File: backend/proyecto/proyecto/src/main/java/com/cibertec/proyecto/service/SimulationService.java
package com.cibertec.proyecto.service;

import com.cibertec.proyecto.dto.request.SubmitAnswerRequest;
import com.cibertec.proyecto.dto.response.*;
import com.cibertec.proyecto.model.Simulacion;

import java.util.List;

public interface SimulationService {
    List<ScenarioSummaryResponse> listarEscenarios();
    Long iniciarSimulacion(Long usuarioId, Long escenarioId);
    List<QuestionResponse> obtenerPreguntas(Long escenarioId, Long simulacionId);
    FeedbackResponse enviarRespuesta(SubmitAnswerRequest request);
    SimulationResultResponse obtenerResultado(Long simulacionId);
    List<Simulacion> listarSimulacionesPorUsuario(Long usuarioId);
    List<Simulacion> listarTodasLasSimulaciones();
}

File: frontend/src/app/services/simulation.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SimulationService {
  private http = inject(HttpClient);
  private apiUrl = 'http://localhost:8080/api/simulaciones';

  listarEscenarios(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/escenarios`);
  }

  iniciarSimulacion(usuarioId: number, escenarioId: number): Observable<number> {
    return this.http.post<number>(`${this.apiUrl}/iniciar?usuarioId=${usuarioId}&escenarioId=${escenarioId}`, {});
  }

  obtenerPreguntas(escenarioId: number, simulacionId?: number): Observable<any[]> {
    const url = simulacionId 
      ? `${this.apiUrl}/preguntas/${escenarioId}?simulacionId=${simulacionId}`
      : `${this.apiUrl}/preguntas/${escenarioId}`;
    return this.http.get<any[]>(url);
  }

  enviarRespuesta(request: any): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}/responder`, request);
  }

  obtenerResultado(simulacionId: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/resultado/${simulacionId}`);
  }

  listarSimulacionesPorUsuario(usuarioId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/usuario/${usuarioId}`);
  }

  listarTodasLasSimulaciones(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/todas`);
  }
}
